core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -94,7 +107,7 @@ object Queue {
       * @param queue    Queue, expressed as vector for fast cons/uncons from head/tail
       * @param deq      A list of waiting dequeuers, added to when queue is empty
       */
-    case class State(queue: Vector[A], deq: Vector[Async.Ref[F,A]])
+    case class State(queue: Vector[A], deq: Vector[Either[Async.Ref[F,A], Async.Ref[F,Vector[A]]]])
```

Comment:
As mentioned above, you end up having to take all the queue contents, and push back whatever you don't need. I can rewrite to do this, but not sure it's really better.